mantidproject/mantid
# Make masked data not contribute to integral in IntegrateMDHistoWorkspace

Closes #14626 

**To Test:**

Use the script below to create a workspace with three peaks. One of the peaks is masked in `masked_ws`.
Compare 1D plots of `masked_ws` and `unmasked_ws`. You can do this by right-clicking on the workspace in the dock and selecting "Plot MD", default settings are fine.
One peak should clearly be missing from `masked_ws` as the masked values have not contributed to the integral.

``` python
# Create a workspace with three peaks
mdws = CreateMDWorkspace(Dimensions=3, Extents=[-10,10,-10,10,-10,10], Names='A,B,C',Units='U,U,U')
FakeMDEventData(InputWorkspace=mdws, PeakParams=[100000,-5,0,0,1])
FakeMDEventData(InputWorkspace=mdws, PeakParams=[100000,0,0,0,1])
FakeMDEventData(InputWorkspace=mdws, PeakParams=[100000,5,0,0,1])
# Bin to give 3 unintegrated dimensions
unmasked_ws = BinMD(mdws,AlignedDim0='A,-10,10,100',AlignedDim1='B,-10,10,100',AlignedDim2='C,-10,10,100')
masked_ws = CloneMDWorkspace(unmasked_ws)
# Mask one of the peaks
MaskMD(Workspace=masked_ws,Dimensions="A,B,C",Extents="-10,-3,-10,10,-10,10")
#Integrate out 2 dimensions, compare 1D plots for the masked and unmasked case
masked_ws = IntegrateMDHistoWorkspace(InputWorkspace=masked_ws, P1Bin=[], P2Bin=[-2,2], P3Bin=[-5,5])
unmasked_ws =IntegrateMDHistoWorkspace(InputWorkspace=unmasked_ws, P1Bin=[], P2Bin=[-2,2], P3Bin=[-5,5])
```
